### ISSUE TITLE
Buttons do not format properly when placed in columns

### ISSUE THREAD

**Issue by gmusser** [WordPress/gutenberg#8440]
**Describe the bug**
When I use to column module to create a row of four buttons, the button in column #1 is not colored as specified.

**To Reproduce**
Add a column module.
Add a button to each column.
Select a background color.

**Expected behavior**
The button should be colored in the editor window as specified, and it should appear as desired in the preview. In actuality, the button is uncolored in the editor, and the next is cropped in the preview.

**Screenshots**
![preview1](https://user-images.githubusercontent.com/40254811/43645542-3de3a1a2-9700-11e8-9ede-96396c391fcc.jpg)
![editor1](https://user-images.githubusercontent.com/40254811/43645551-4474fef8-9700-11e8-8737-caccbb37f147.jpg)

**Desktop (please complete the following information):**
Mac OS 10.11.6, Safari 11.1.2

**Comment by aduth** [comment]
There appears to be a bug with button background colors generally, not specific to Columns. Investigating...

> and the next is cropped in the preview.

Can you elaborate on what you mean by cropped?

**Comment by aduth** [comment]
>There appears to be a bug with button background colors generally, not specific to Columns. 

This turned out to be an issue for me only with my specific theme, which was not applying color styling correctly (the "Gutenberg Starter" theme). Can you clarify which theme you are using? I'm not able to reproduce this issue under the Twentyseventeen theme.

**Comment by gmusser** [comment]
Sorry, I meant "text",  not "next".  By "cropped" I meant that the word is cutoff - not all the letters that I entered in the editor are shown.

**Comment by spencerfinnell** [comment]
I've had this happen a few times too but can't consistently reproduce it. Editing the block visually shows the correct text, but switching to the HTML editor shows the incorrect text. Correcting it in the HTML fixes it on the frontend.

**Comment by aduth** [comment]
@spencerfinnell Which browser are you using when encountering the issue?

**Comment by spencerfinnell** [comment]
Latest Chrome on a couple different macOS devices -- both Sierra and High Sierra

**Comment by spencerfinnell** [comment]
It also was not in a column, nor did it have a custom color applied.

**Comment by aduth** [comment]
Thanks for the additional context. I think I've tracked down a reliable steps to reproduce:

1. Insert button block
2. Press ArrowUp to focus the button text field
3. Type a single letter
4. Press backspace _twice_
5. Type a few letters
6. Press Preview
7. Note only the first of the letters typed in Step 5 is shown

**Comment by aduth** [comment]
Fix for the cropping issue at #8451

I'd still appreciate more information about what's going on with the background colors and whether this may be attributable to an incompatibility with the specific theme being used.

**Comment by aduth** [comment]
>>There appears to be a bug with button background colors generally, not specific to Columns.
>
>This turned out to be an issue for me only with my specific theme, which was not applying color styling correctly (the "Gutenberg Starter" theme).

Observing that this is tracked at https://github.com/WordPress/gutenberg-starter-theme/issues/71

**Comment by spencerfinnell** [comment]
It does appear to be theme-specific. I can reproduce with your steps only on one theme I'm testing. Looking for the inconsistency.

**Comment by spencerfinnell** [comment]
Actually it's not. It only is reproducible with those steps when `SCRIPT_DEBUG` is on.

```
react-dom.24169eaf.js:526 Warning: Unsafe lifecycle methods were found within a strict-mode tree:
    in Unknown (created by e)
    in e (created by RemountOnPropChange(e))
    in RemountOnPropChange(e)
    in WithSelect(Component)

componentWillReceiveProps: Please update the following components to use static getDerivedStateFromProps instead: t

componentWillUpdate: Please update the following components to use componentDidUpdate instead: t
```

**Comment by aduth** [comment]
Considering this as closed by #8440, which addresses issues with cropped characters.

If issues persist with coloring, please verify that the theme has applied any custom Gutenberg colors correctly. If it's not an issue with theme, we can reopen or create a new issue.
